src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb

```diff
@@ -3927,14 +3971,22 @@ class C
                 Await state.WaitForAsynchronousOperationsAsync()
 
                 Assert.Equal("#", state.GetLineFromCurrentCaretPosition().GetText())
-                Await state.AssertNoCompletionSession()
+                If completionImplementation = CompletionImplementation.Legacy Then
+                    Await state.AssertNoCompletionSession()
+                Else
+                    Await state.AssertCompletionSession() ' Agreed on changing the behavior
```

Comment:
who agreed?  when/when/why?  Changing completion behavior is extremely destabalizing and frustrating to users.

Reply:
The editor team and the Roslyn team agreed on this. We decided that the change is an improvement not a regression. We think it would better serve user needs.

---
In reply to: [222523306](https://github.com/dotnet/roslyn/pull/29016#discussion_r222523306) [](ancestors = 222523306)

Reply:
If this is better behavior, then we shoudl change the legacy behavior as well.

Reply:
I tried to fix the issue in the old completion. I could not find an easy fix: we should dismiss the session, if used manually (arrow keys) changes the caret but we should not do this if the caret is change automatically. 

If you know how to fix it in the original code, please let me know. If not, I suggest to keep the old behavior as is.

---
In reply to: [222850452](https://github.com/dotnet/roslyn/pull/29016#discussion_r222850452) [](ancestors = 222850452)

Reply:
> we should dismiss the session, if used manually (arrow keys) changes the caret but we should not do this if the caret is change automatically.

:confused: I don't understand how this reply relates to this code

I also agree this should be fixed in the legacy implementation, preferably as a separate pull request merged *before* this pull request and then accounted for in this pull request.

Reply:
I already tried to fix this in the legacy implementation. It is not trivial. My suggestion is to skip this. It should not block anybody.

---
In reply to: [229264505](https://github.com/dotnet/roslyn/pull/29016#discussion_r229264505) [](ancestors = 229264505)

Reply:
The scenario is typing "#" on
class C
{
    $$
}

Both old and new completions move the caret to the left but 
- the old completion dismisses the completion session(caret position change)
- the new completion keeps the completion session (caret position change made automatically not by human)

I tried to repair this behavior in the old completion but I could not separate this scenario where user manually moves the caret: we need to dismiss session in the case.

---
In reply to: [230989526](https://github.com/dotnet/roslyn/pull/29016#discussion_r230989526) [](ancestors = 230989526)

Reply:
> I tried to repair this behavior in the old completion but I could not separate this scenario where user manually moves the caret: we need to dismiss session in the case.

Why not have the new completion just have the behavior of the old completion?

Reply:
We cannot subscribe for the Editor's OnCaretPositionChanged in the new model. 
Since we do not consider this change as a regression, the editor does not want to expose OnCaretPositionChanged.

---
In reply to: [232508924](https://github.com/dotnet/roslyn/pull/29016#discussion_r232508924) [](ancestors = 232508924)